sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -36,14 +37,20 @@
             .identityClientOptions(identityClientOptions)
             .build();
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
-        if (configuration.contains(Configuration.PROPERTY_MSI_ENDPOINT)
-                || (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)
-                        && configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER))) {
-            appServiceMSICredential = new AppServiceMsiCredential(clientId, identityClient);
-            virtualMachineMSICredential = null;
+        if (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)) {
+            if (configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER)) {
+                if (configuration.contains(PROPERTY_IDENTITY_SERVER_THUMBPRINT)) {
+                    managedIdentityServiceCredential = new ServiceFabricMsiCredential(clientId, identityClient);
+                } else {
+                    managedIdentityServiceCredential = new AppServiceMsiCredential(clientId, identityClient);
+                }
+            } else {
+                managedIdentityServiceCredential = null;
```

Comment:
Similar to the comments in PR #13466, throw the exception here.

Reply:
same response.